src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,215 @@
+        // Strict-Transport-Security: max-age=<expire-time>; includeSubDomains
+        // Strict-Transport-Security: max-age=<expire-time>; preload
+        // This function accomplishes the following:
+        // "max-age=31536000; includesubdomains; preload" => {"max-age":31536000,"includesubdomains":true,"preload":true}
+        const parse = (headerValue: string) => {
+            const parsedHeader = {};
+            const directives = headerValue.toLowerCase().split(';');
+            const nameValuePairRegex = /^ *([!#$%&'*+.^_`|~0-9A-Za-z-]+) *= *("(?:[~0-9])*"|[!#$%&'*+.^_`|~0-9]+) *$/;
+            // Regex for name-value pairs. E.g.: max-age=31536000
+            // Modified usage of https://github.com/jshttp/content-type/blob/64bde0d996ccb4334341662c0c7d25f7b370c4d9/index.js#L23
+            const tokenRegex = /^ *[!#$%&'*+.^_`|~0-9A-Za-z-]+$/;
+            // Regex for single tokens. E.g.:  includesubdomains
+
+            directives.forEach((directive) => {
+                const match = tokenRegex.exec(directive) || nameValuePairRegex.exec(directive);
+
+                if (!match) {
+                    throw new Error(`'strict-transport-security' header has the wrong format: ${directive}`);
+                }
+
+                const [matchString, key, value] = match;
+                const name = key || matchString.trim();
+
+                if (parsedHeader[name]) {
+                    throw new Error(`'strict-transport-security' header contains more than one '${name}'`);
+                }
+
+                parsedHeader[name] = value || true;
+            });
+
+            return parsedHeader;
+        };
+
+        const isInvalidValue = (maxAge: string, limit: number): boolean => {
+            return maxAge && parseInt(maxAge) < limit;
+        };
+
+        const verifyPreload = (resource, parsedHeader) => {
+            const hostname = url.parse(resource).hostname;
+
+            const match = _.find(preloadList, { name: hostname }) || _.find(preloadList, { name: hostname.replace('www.', '') });
+
+            if (match && !parsedHeader.preload) {
+                throw new Error(`This site is in the preload list, but 'Strict-transport-security' header doesn't have 'preload' attribute.`);
+            }
+
+            if (match && !parsedHeader.includesubdomains) {
+                throw new Error(`This site is in the preload list, but 'Strict-transport-security' header doesn't have 'includesubdomains' attribute.`);
+            }
+
+            if (!match && parsedHeader.preload) {
+                throw new Error(`This site is not in the preload list, but 'Strict-transport-security' header has 'preload' attribute`);
+            }
+        };
+
+        const validate = async (fetchEnd: IFetchEnd) => {
+            const { element, resource, response }: { element: IAsyncHTMLElement, resource: string, response: IResponse } = fetchEnd;
+
+            // This check does not make sense for data URI.
```

Comment:
Also for `javascript:` and others (see: [this](https://github.com/sonarwhal/sonar/commit/28c782db16ca6140a083fae76f143a05f595694e#diff-3f3def15d620b7145d22d1a373aa59f0R87)).